Add privacy policy (#10)
mello-app/static-website: privacy.md

---
layout: main
title: "Privacy Policy"
---

Your privacy is just as important (if not more) than our privacy. We go to big lengths to ensure that your privacy is protected as much as possible. We're commited to making sure our service is private-by-default and we're a member of the <a href="https://doamatto.xyz/projects/privacy-by-default">Private by Default campaign.</a>

### What data is collected and why?

> When you download the application, you need to make an account. If you sign up with Apple, Apple gives us the following data:

Reviewer: It's probably better to make it seem that we aren't receiving any data ("give" might have that connotation), so saying that Apple is simply allowing access fixes that issue.
```suggestion
When you download the application, you need to make an account. If you sign up with Apple, Apple will allow us access to the following data:
```

> - your first and last name (we ignore this data); and:
> - your email address (we ignore this data, as well)

Reviewer: Seems unclear to specify this for all the types of data that we are allowed access to, would be best to simply tell the user a) what data we could access and b) what we do with it
```suggestion
- your first and last name
- your email address
Even though Apple allows us access to the above-mentioned, we do not access or use any of this data when you access Mello.
```

Reviewer: Also, just from reading this, it's unclear specifically how we do authentication with Apple if we do not use any of the data provided? Is there a token or unique identifier that they also provide? If there is, it's best that we add that to the privacy policy under what's collected.

Author: Apple does provide us with an access token of sorts for OAUTH that's meant to be used in place of the hash you would for normal logins (afaik). It's documented [here](https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_rest_api/authenticating_users_with_sign_in_with_apple); I'll try to better document how we use this token when I get a chance

Author: Clarified in 23de3fe

If you sign up with an email address, we collect the following data:

> - your email address (used for login and password recovery)
> - a password that you provide (used for login)

Reviewer: Would be best to change wording from password "recovery" to "resets", to make it clear that passwords are not stored in their original form. To add on to that it's would also be nice to add on to the password entry by clarifying that it's not stored but instead hashed securely.
```suggestion
- your email address (used for logging in and password resets)
- a password that you provide (used for logging in, securely hashed and salted)
```

Reviewer: did not mean to make this a review

When using your account, you can choose to favourite meditations, sleep stories, and other content on Mello. When you favourite one of these, the following data is stored with your account data:

> - the content ID of what you favourited (used to store your favourites for whenever you need them)

Author: Describe how this is stored with CloudKit when using Sign In With Apple

### What data is shared, sold, or transfered?
We do not sell your data with any third-parties. [Some data is shared with third-parties.](#who-has-access-to-or-processes-your-data)

### Who has access to or processes your data?
[Service X](#), our database provider, has access to:
- salted and hashed emails (if you sign up with your email address. You can avoid this by signing in with Apple)
- salted and hashed passwords
- salted and hashed content IDs (if you favourite content. You can avoid this by not using the favourite functionality)

> If you sign in with [Apple](https://www.apple.com/legal/privacy/en-ww/), you are under their privacy policy. Review to know what you're getting yourself into.

Author: Fix wording to be clearer

> If you sign in with [Apple](https://www.apple.com/legal/privacy/en-ww/), you are under their privacy policy. Review to know what you're getting yourself into.

Reviewer: Best to remove or adjust the last sentence, seems slightly snarky. Would also be good to move the privacy policy link to actually be for the words "privacy policy".
```suggestion
If you sign in with Apple, you are under their [privacy policy](https://www.apple.com/legal/privacy/en-ww/). Review to know their data policies.
```
Or optionally:
```suggestion
If you sign in with Apple, you are under their [privacy policy](https://www.apple.com/legal/privacy/en-ww/).
```

> The U.S. government could request information from us. We will only provide data if they provide us a legal order such as a subpoena. When they provide us with such order and an email address to find the user in question, we can't give them any information because:

Reviewer: A user would probably feel slightly better if we explain why the U.S. government has the authority to request user data.
```suggestion
Mello, it's servers, and it's members are under the jurisdiction of the United States. This means that the U.S. government has the ability to request user information from us. Due to our commitment to privacy, we would only provide data if they provide us with a legal order such as a subpoena. Should they provide us with such order and a valid email address to find the user in question, we will still be unable to give them any information because:
```

- we can't unsalt and unhash content IDs, and:
- we can't unsalt and unhash passwords.

### What if there's a data breach?

> In the event of a data breach, we will send out an email address, regardless of if you were affected or not, with password reset notifications, following information pertaining to what things may have been compromised. This will all be done as soon as we find said breach. Once this issue is patched, we may release information on the issue as well as how we fixed it on GitHub.

Reviewer: Grammar fixes
```suggestion
In the event of a data breach, we will send out an email notification, regardless of if you were affected or not, with password reset notifications, following information pertaining to what things may have been compromised. This will all be done as soon as we find said breach. Once this issue is patched, we may release information on the issue as well as a write-up on how we fixed it on GitHub.
```

### Your rights and choices
In some countries, states, or provinces, you may have certain rights pertaining to your data. Regardless of where we're from, we want to give you as many choices as possible.

> **TODO: See if there are other things that need to go here**

Author: Resolve TODO section ( needs researching )

Reviewer: GDPR also requires simply being able to request and view all available data on yourself, if you feel the need to add that

Author: Added request to view in f1f2243

When you download the application, you need to make an account. If you sign up with Apple, Apple will allow us access to the following data:
- your first and last name (we ignore this data),
- your email address (we use this for data breach notifications); and:
- a unique token for us for authentication (we use this in place of an email address or password).

> If you sign up with an email address, we collect the following data:

Reviewer: ```suggestion
Alternatively, if you sign up with an email address, we collect the following data:
```
References last option, reinforcing that there are two options

- your email address (used for logging in, password resets, and data breach notifications),

> - a unique token for us for authentication (we use this in place of an email address or password).

Reviewer: This does not make sense given that it's using an email address. Are you referring to sending an email to the user every time they want to login and authenticating them through a link attached in the email?

Author: Exactly; you get an email from Magic with a one-time link that expires after 10 minutes to login.

Reviewer: Seems good.

When using your account, you can choose to favourite meditations, sleep stories, and other content on Mello. When you favourite one of these, the following data is stored with your account data:
- the content ID of what you favourited (used to store your favourites for whenever you need them)

> Your settings that you use in the app will automatically sync between devices to give a seamless experience. This is stored with your content ID 

Reviewer: ```suggestion
Your settings that you use in the app will automatically sync between devices to give a seamless experience. This is stored alongside your favourited content IDs.
```
"with your content ID" is singular, might be slightly confusing / vague

### What data is shared, sold, or transfered?
We do not sell your data with any third-parties. [Some data is shared with third-parties](#who-has-access-to-or-processes-your-data) so that we may provide our service.

### Who has access to or processes your data?
[Magic](https://magic.link), our login provider, has access to:
- email addresses (either the one provided by you or Apple),

> - your first and last name (if you signed in with Apple).

Reviewer: 1. Is this data required for us / Magic to collect?
2. If not, does Apple automatically share this data or can we prevent ourselves / Magic from collecting it

Author: 1. Email addresses do need to be collected to generate hashes for users and allow for authentication
2. Apple, just like how we get passed someone's name and email through Sign in with Apple, passes that data through Magic to allow for auth. It may be possible to tell users prior that they might want to use fake information for their name and the "Hide my email" functionality, but it doesn't seem practical

Reviewer: I see, well if Magic collects it automatically then it makes sense

### How do we protect your data?
**TODO: Elabourate on what else we do to protect data.**

> Truth be told: there's no be all and end all to protect your data. But, we do our best to use modern-day encryption methods as well as strict HSTS, DNSSEC, and CSP to ensure your experience is the way it was intended. We hash passwords with Argon2 locally, before sending them off your device. We also give you the choice to become independent of an account to keep all your data on your device.

Reviewer: Probably best to state this fact somewhere in the beginning of the document then just talk about "what if" the user has an account with us, and what that means for their data

Author: This has now been adjusted to be in the talks of data collection in the first place.
